src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeBuilderImpl.cs

```diff
@@ -564,11 +718,184 @@ public override ConstructorInfo[] GetConstructors(BindingFlags bindingAttr)
         [DynamicallyAccessedMembers(GetAllMembers)]
         public override MemberInfo[] GetMember(string name, MemberTypes type, BindingFlags bindingAttr) => throw new NotSupportedException();
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2085:DynamicallyAccessedMembers", Justification = "Methods are loaded from this TypeBuilder")]
         public override InterfaceMapping GetInterfaceMap([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods)] Type interfaceType)
-            => throw new NotSupportedException();
+        {
+            ThrowIfNotCreated();
+            ArgumentNullException.ThrowIfNull(interfaceType);
+            ValidateInterfaceType(interfaceType);
+
+            InterfaceMapping im = default;
+            im.InterfaceType = interfaceType;
+            im.TargetType = this;
```

Comment:
I would initialized this completely in one pass as opposed to splitting it before and after the `GetMethods()` below. I don't know the type, but it might be possible to initialize this in a manner without the `default`.

Reply:
Not sure how to do that, this is a struct without any constructors that setting these fields
https://github.com/dotnet/runtime/blob/6de7549ae1872009baade9091be76071f955c47d/src/libraries/System.Private.CoreLib/src/System/Reflection/InterfaceMapping.cs#L6-L12